### DISCUSSION TITLE
500 viewers crashes stream

### DISCUSSION THREAD

**Discussion by articks** [bluenviron/mediamtx#3024] [Questions]
### Question

If there is only one stream and there are 500 people watching, the video and audio begin to slow down and crash. The input RTMP stream is converted to WebRTC using ffmpeg, otherwise the sound does not work.

```
"~^rtmp/([0-9a-z]+)$":
    runOnReady: ffmpeg -i rtmp://hostname.tld:1935/$MTX_PATH -vcodec copy -c:a libopus -f rtsp rtsp://hostname.tld:$RTSP_PORT/$G1
    runOnReadyRestart: yes
```

Maybe I'm doing something wrong? Is there an easier way to convert RTMP to WebRTC so that the sound works? And how to organize support for a large number of viewers in this case? So far, the only things that come to mind are a load balancer and proxying to multiple servers.

Need help!

**Comment by aler9** [comment]
There is no limit on 500 streams, the limit is certainly dictated by your specific LAN card and network throughput between server and readers. You didn't provide logs or anything else that allows to understand the issue, therefore nothing else can be inferred.

If you want to scale, my advice is to find the bottleneck by monitoring respectively incoming and outgoing packets, CPU and RAM. If the bottleneck lies in the available bandwidth, as everything suggests, you need to upgrade your network capability.

If you still have questions please open another discussion and provide a detailed description of the issue including setup, logs, name of software that is publishing, name of software that is reading, as explicitly mentioned in the instructions.
